### ISSUE TITLE
Number Server: Provide RDF relations for dclp-hybrid and TM No.

### ISSUE THREAD

**Issue by Edelweiss** [DCLP/dclpxsltbox#144]
Number Server needs to produce RDF like this:

``` xml
<rdf:RDF>
  <rdf:Description rdf:about="http://dclp.atlantides.org/dclp/100152/source">
    <dc:relation rdf:resource="http://dclp.atlantides.org/dclp/p.oxy;61;4098/source"/>
  </rdf:Description>
</rdf:RDF>
```

There are two kinds of dclp identifiers, one that comes along with a TM No. and one giving the name, volume number and document number of the respective publication, which is called dclp-hybrid idno in the EpiDoc files' publication statements.

```
dclp/<TM No.>
dclp/<Series;Volume;Number>

dclp/100152
dclp/p.oxy;61;4098
```

The Number Server needs to learn to distinguish the two.

Since a huge mass of dclp data doesn't have a dclp hybrid, the number server won't be able to establish a relationship between publication number and TM No. for those files.

Later in the editor these two identifier classes will be called:
- **DCLP/TM  No.** (here a dropdown of TM numbers will appear)
- **DCLP Publication** (here three dropdown boxes for the abbreviated name of the series, the volume number and the document number will appear)

In SoSOL's section »Existing publication« on the page for »Advanced Creation« there shall be a link to LDAB: »To find the appropriate TM number search [here](http://www.trismegistos.org/ldab/search.php)«

**Comment by rla2118** [comment]
Carmen,

Can we have this »To find the appropriate TM number search here
<http://www.trismegistos.org/ldab>« point rather to this page:
http://www.trismegistos.org/ldab/search.php?

Rodney

**Comment by Edelweiss** [comment]
I have setup an xslt that can be called up by the mapping process’ [clojure](https://github.com/papyri/navigator/blob/dclp/pn-mapping/src/info/papyri/map.clj) file that is responsible for the creation of the rdf files.

If you pair up a given DCLP EpiDoc file, such as [P.Oxy 61 1015](https://github.com/DCLP/idp.data/blob/dclp/DCLP/101/100152.xml), with the [rdf xslt](https://github.com/papyri/navigator/blob/dclp/pn-mapping/xslt/dclp-rdf.xsl) for DCLP files, they will sire the following output:

```xml
<rdf:Description xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#"
                 xmlns:dct="http://purl.org/dc/terms/"
                 xmlns:foaf="http://xmlns.com/foaf/0.1/"
                 xmlns:rdfs="http://www.w3.org/2000/01/rdf-schema#"
                 rdf:about="http://dclp.atlantides.org/dclp/100152/source">
   <dct:identifier>dclp.atlantides.org/dclp/100152</dct:identifier>
   <dct:identifier>dclp.atlantides.org/ldab/10801</dct:identifier>
   <dct:identifier>dclp.atlantides.org/dclp/p.oxy;61;4098</dct:identifier>
   <dct:source>
      <rdf:Description rdf:about="http://dclp.atlantides.org/dclp/p.oxy;61;4098/work">
         <dct:source rdf:resource="http://dclp.atlantides.org/dclp/p.oxy;61;4098/original"/>
      </rdf:Description>
   </dct:source>
   <dct:identifier>100152</dct:identifier>
   <dct:isPartOf>
      <rdf:Description rdf:about="http://dclp.atlantides.org/dclp/p.oxy;61">
         <rdf:type rdf:resource="http://purl.org/ontology/bibo/Book"/>
         <dct:isPartOf>
            <rdf:Description rdf:about="http://dclp.atlantides.org/dclp/p.oxy">
               <rdf:type rdf:resource="http://purl.org/ontology/bibo/Series"/>
               <dct:isPartOf rdf:resource="http://dclp.atlantides.org/dclp"/>
            </rdf:Description>
         </dct:isPartOf>
      </rdf:Description>
   </dct:isPartOf>
   <dct:relation rdf:resource="http://www.trismegistos.org/text/100152"/>
   <dct:relation rdf:resource="http://www.trismegistos.org/ldab/text.php?quick=10801"/>
   <foaf:page>
      <rdf:Description rdf:about="http://dclp.atlantides.org/dclp/p.oxy;61;4098">
         <foaf:topic rdf:resource="http://dclp.atlantides.org/dclp/100152/source"/>
      </rdf:Description>
   </foaf:page>
</rdf:Description>
```

There may additional desiderata as for what the output should look like (for instance how to represent LDAB identifiers, how to handle reprints, how to parse page information from a given dclp-hybrid identifier etc. I also interspersed a few comments on these matters into the source file itself), but for a start the rdf comprises the xpaths where SoSOL is looking for mapping information in order to create a new publication for editing.

But...

**Comment by Edelweiss** [comment]
...this is only half the loaf.

Of course we also need the counterpart to this file, which should look like this:

```xml
<rdf:RDF>
  <rdf:Description rdf:about="http://dclp.atlantides.org/dclp/p.oxy;61;4098/source">
    <dc:relation rdf:resource="http://dclp.atlantides.org/dclp/100152/source"/>
  </rdf:Description>
</rdf:RDF>
```
(It's the same example as above, just the two identifiers swapped)

The xslt part of the job can be accomplished within a couple of minutes, but we have to consider how it can be done best:

* Should the rdf xslt simply produce another rdf:Description tag if dclp-hybrid information is present? This will leave the rdf snippet in an invalid xml state because will have two root elements. But perhaps this is ok, as an intermediate step.
* Do we want to have an extra xslt for this purpose? - That would look ugly, wouldn't it?
* Do we want to provide a parameter to the current file in order to be able to call it for different purposes? - I would incline towards this idea, but of course the clojure scripts needs to learn how to deal with that.

I think we can come to a decision later. As I said, we already have half the loaf, which is better than nix on the table. This will allow us to call up a publication by its PN/DCLP identifier of the form **dclp.atlantides.org/dclp/100152**.

**Comment by Edelweiss** [comment]
I just noticed that I have been committing to the wrong repository. So I merged the new xslt file from papyri’s dclp branch to dclp’s xslt-master branch:

[dclp-rdf.xsl](https://github.com/DCLP/navigator/blob/xslt-master/pn-mapping/xslt/dclp-rdf.xsl)

**Comment by jcowey** [comment]
Closed for now as through #188 this is probably no longer applicable
